sim_pipeline/gg_lens.py

```diff
@@ -5,6 +5,34 @@
 from lenstronomy.LensModel.Solver.lens_equation_solver import LensEquationSolver
 
 
+def image_separation_from_positions(image_positions):
+    """
+    calculate image separation in arc-seconds
+
+    :param image_positions: list of image positions in arc-seconds
+    :return: image separation in arc-seconds
+    """
+    if len(image_positions[0]) == 2:
+        image_separation = np.sqrt((image_positions[0][0] - image_positions[0][1]) ** 2 + (
+                image_positions[1][0] - image_positions[1][1]) ** 2)
+    else:
+        coords = np.stack((image_positions[0], image_positions[1]), axis=-1)
+        separations = np.sqrt(np.sum((coords[:, np.newaxis] - coords[np.newaxis, :]) ** 2, axis=-1))
+        image_separation = np.max(separations)
+    return image_separation
+
+
+def theta_e_when_source_infinity(deflector_dict):
```

Comment:
I would recommend having as input vel_disp directly not relying on a dictionary with a specific keyword argument. This way you can also specify that the input needs to be in km/s.